src/index.ts

```diff
@@ -0,0 +1,24 @@
+///<reference path="../typings/index.d.ts" />
+import { EscapeStack } from './EscapeStack';
+import { handleEscapeKeydown } from './domEventHandler';
+
+let globalStack;
+
+const es = new EscapeStack();
+handleEscapeKeydown(document.body, es.pop);
+
+function createEscapeStack(global: boolean = false) {
+    const es = new EscapeStack();
+    handleEscapeKeydown(document.body, es.pop);
```

Comment:
It looks like this behavior is a little different from the old logic:

```javascript
module.exports = function(global) {		
    if (!global) {		
        return makeStack();		
    }		
		
    if (!globalStack) {		
        globalStack = makeStack();		
    }		
    return globalStack;		
};
```

In the old logic, a new `EscapeStack` was created only when global was false, but here it looks like it happens every time, so there's a chance that this new object is never actually used.

You can probably just refactor these 2 lines into a `makeStack()` function and then use the exact same code as before.